src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -557,6 +587,11 @@ private void AttributePrefix(IEnumerable<HtmlSymbol> whitespace, IEnumerable<Htm
                 // Output the attribute name, the equals and optional quote. Ex: foo="
                 Output(SpanKind.Markup);
 
+                if (quote == HtmlSymbolType.Unknown && whitespaceAfterEquals.Any())
```

Comment:
When would this evaluate to `false`? You're in an `else` block for `quote != HtmlSymbolType.Unknown || !whitespaceAfterEquals.Any()`. Do a bit of Boolean algebra on `!(A != B || !C)` and you get `A == B && C`.

Looks line [line 594](https://github.com/aspnet/Razor/pull/503/files#diff-58782e50bed88f4f2f70af4835346999R595) is unreachable. What is lost if that code is never evaluated?